redis/src/aio/connection.rs

```diff
@@ -386,6 +386,7 @@ pub(crate) async fn connect_simple<T: RedisRuntime>(
     Ok(match connection_info.addr {
         ConnectionAddr::Tcp(ref host, port) => {
             let socket_addrs = get_socket_addrs(host, port).await?;
+            #[allow(clippy::manual_try_fold)]
```

Comment:
I think the two instances in this file should probably be refactored to not use a fold, honestly. I don't think there's any reason to evaluate all the potential addresses, so we can probably just do something like this instead:

```
future::select_ok(
    socket_addrs.into_iter().map(|socket_addr| {
        Box::pin(async move { <T>::connect_tcp(socket_addr).await })
    })).await?.0
```

Wanna try it?